src/PolarNightMidnightSunResolver.js

```diff
@@ -0,0 +1,56 @@
+import { isValidDate } from './DateUtils';
+import adhan from './Adhan';
+
+export const PolarNightMidnightSunResolver = {
```

Comment:
the name is a bit verbose, this phenomenon is something that occurs inside the polar circle, so perhaps we can go with `PolarCircleResolution`

Reply:
Excellent, I will rename it.